internal/cloudformation/tasks/upsert.go

```diff
@@ -88,11 +88,34 @@ func upsertStack(
 	checkError(err)
 
 	if *changeSet.Status == "FAILED" {
-		printer.Fatal(
-			fmt.Errorf("Cloudformation ChangeSet failed to create"),
-			*changeSet.StatusReason,
-			"",
-		)
+		if *changeSet.StatusReason == "The submitted information didn't contain changes. Submit different information to create a change set." {
+			printer.Error(
+				fmt.Errorf("Cloudformation ChangeSet failed to create: no changes"),
+				*changeSet.StatusReason,
+				"",
+			)
+			printer.SubStep("Cleaning up unused change set", 1, true, true)
+			_, err := cf.DeleteChangeSet(
+				&awsCF.DeleteChangeSetInput{
+					ChangeSetName: changeSet.ChangeSetId,
+				},
+			)
+			if err != nil {
+				printer.Fatal(
+					err,
+					"Manually clean up change set",
+					"",
+				)
+			}
+			printer.Stop()
+			os.Exit(1)
```

Comment:
Can we please not call `os.Exit(1)` directly? It complicates testing. Use `printer.Fatal`, or refactor the function to return an error, and let it cascade back out of the testable path.

Reply:
Yeah good pickup, this was a quick fix, not ideal to merge in like that.

Reply:
After double checking, I've noticed that there's more `os.Exit` calls scattered around the place. I think I'll make a pass at trying to refactor those out.